kafka-client/src/main/java/dev/responsive/kafka/api/KeyValueStoreOptions.java

```diff
@@ -0,0 +1,92 @@
+package dev.responsive.kafka.api;
+
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.state.KeyValueStore;
+
+/**
+ * Configuration options for building a Responsive {@link KeyValueStore}. Use {@link #named(String)}
+ * (or {@link #namedGlobal(String)} if this is a global store) to get a new object and pass that in
+ * to one of the {@link ResponsiveStoreFactory} APIs, or apply any of the additional optional
+ * configs via the {@code #withXXX} methods below.
+ */
+public class KeyValueStoreOptions<K, V> {
+
+  // required configs
+  private final String name;
+  private final boolean isGlobal;
+
+  // optional configs
+  private boolean truncateChangelog = false;
+  private Long ttlMs = null;
```

Comment:
what lifetime does this refer to (all keys? change log topic? changelog topic tombstones?)

Reply:
Well technically those should all be the same, no? (with maybe a little extra time added as a buffer for the actual changelog retention, like what Streams does with the `windowstore.additional.changelog.ms` config or whatever it's called)

I didn't actually add any setters for this so it's not exposed at the moment anyway, but if we're thinking of offering/exposing this in the near future it would be nice to include it when laying the groundwork so it's easy to switch on when ready

Of course this is kind of getting ahead of things so I can take it out 🤷‍♀️

Reply:
Lol are you seeing the emoji come up as "🤷  ♀️ "? Looks like the gender-based emojis broke (it the previous it shows up correctly as the woman-shrugging) 😂

Reply:
> Well technically those should all be the same, no?

Not following. Each of these seem like different properties to me:
- Record TTL: A semantic property of the store that defines a lifetime for each entry (in stream time)
- Retention Time: An implementation detail of a store that defines how long records are stored (in wall clock time)
- Changelog retention: A property of the changelog Kafka topic (for topics that use the `delete` policy). We could still set this for topics with infinite Record TTL if the user is willing to accept data loss if no flush happens for tha time period.
- Tombstone retention: A property of the changelog Kafka topic (for topics that use the `compact` policy)
- (Related but not exactly TTL) Cleanup policy: the cleanup policy of the Kafka topic

I'm not suggesting we expose or not expose all these as configurable by the user. I was just wondering which of these the ttlMs refers to. Or maybe it doesn't refer to any of these exactly? Basically - what does this config mean 😄 ?

> Lol are you seeing the emoji come up as "🤷 ♀️ "? Looks like the gender-based emojis broke (it the previous it shows up correctly as the woman-shrugging) 😂

Yep that's what it looks like to me. They should fix that. It's sad that it :shrugs